src/storage/core.rs

```diff
@@ -390,7 +393,10 @@ impl<K: Key + 'static> Storage<K> {
         let possible_blobs = blobs
             .iter_possible_childs_rev(key)
             .map(|(id, blob)| async move {
-                if matches!(blob.data.check_filters(key).await, Ok(true)) {
+                if matches!(
+                    blob.data.check_filter(key).await,
+                    FilterResult::NeedAdditionalCheck
```

Comment:
I'd prefer to inverse this check: compare for inequality with `FilterResult::NoContains`. It would be safer if `FilterResult` will be extended some day